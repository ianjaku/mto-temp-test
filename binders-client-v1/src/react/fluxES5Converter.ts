/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
