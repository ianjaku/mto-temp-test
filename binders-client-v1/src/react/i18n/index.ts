import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
