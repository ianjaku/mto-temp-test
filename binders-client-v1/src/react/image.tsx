/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
