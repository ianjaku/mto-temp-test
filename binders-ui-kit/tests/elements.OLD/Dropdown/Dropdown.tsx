import * as React from "react";
import Dropdown from "../../../src/elements/dropdown";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const TYPE = "testtype";
const ELEMENT1 = { id: 1, label: "element 1"};
const ELEMENT2 = { id: 2, label: "element 2"};
const ELEMENTS = [ ELEMENT1, ELEMENT2];
const AVATAR_ELEMENT1 = { id: 1, label: "account 1", avatar: "http://path.to/image1.jpg"};
const AVATAR_ELEMENT2 = { id: 2, label: "account 2", avatar: "http://path.to/image2.jpg"};
const AVATAR_ELEMENTS = [ AVATAR_ELEMENT1, AVATAR_ELEMENT2];

const getTestElements = () => ELEMENTS;
const getAvatarElements = () => AVATAR_ELEMENTS;


const createDefaultDropdown = (open) => wrapWithTheme(<Dropdown type={TYPE} elements={getTestElements()} open={open}/>);
const createDropup = () => wrapWithTheme(<Dropdown type={TYPE} elements={getTestElements()} />);
const createDropdownWithSelectedElement = () => wrapWithTheme(<Dropdown type={TYPE} elements={getTestElements()} selectedElementId={ELEMENT2.id} />);
const createDropupWithAvatarsAndHorizontalRules = (open) => wrapWithTheme(<Dropdown type={TYPE} elements={getAvatarElements()} selectedElementId={AVATAR_ELEMENT1.id} horizontalRulePositions={[1]} open={open} />);

// snapshots
test("Default dropdown closed (snapshot)", () => {
    const dropdown =  create(createDefaultDropdown(false));
    const serialized = dropdown.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("Default dropdown open (snapshot)", () => {
    const dropdown =  create(createDefaultDropdown(true));
    const serialized = dropdown.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("dropup (snapshot)", () => {
    const dropdown = create(createDropup());
    const serialized = dropdown.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("dropdown with selected elemnent (snapshot)", () => {
    const dropdown = create(createDropdownWithSelectedElement());
    const serialized = dropdown.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("dropdown with avatars and horizontal rules (snapshot)", () => {
    const dropdown = create(createDropupWithAvatarsAndHorizontalRules(true));
    const serialized = dropdown.toJSON();
    expect(serialized).toMatchSnapshot();
});

// enzyme
test("Default dropdown closed (mount: enzyme+jsdom)", () => {
    const ddMount = mount(createDefaultDropdown(false));
    const dropdownFieldLabel = ddMount
        .find(".dropdown-field-label-group-text")
        .first()
        .text();
    expect(dropdownFieldLabel).toEqual(TYPE);
    const elementsListItems = ddMount.find(".dropdown-elements li");
    expect(elementsListItems).toHaveLength(0);
});

// enzyme
test("Default dropdown open (mount: enzyme+jsdom)", () => {
    const ddMount = mount(createDefaultDropdown(true));
    const elementsListItems = ddMount.find(".dropdown-elements li");
    expect(elementsListItems).toHaveLength(ELEMENTS.length);
    const firstElementText = ddMount
        .find(".dropdown-elements li")
        .first()
        .text();
    expect(firstElementText).toEqual(ELEMENT1.label);
});

test("Dropdown with selected element id (mount: enzyme+jsdom)", () => {
    const ddMount = mount(createDropdownWithSelectedElement());
    const dropdownFieldLabel = ddMount
        .find(".dropdown-field-label-group-text")
        .first()
        .text();
    expect(dropdownFieldLabel).toEqual(ELEMENT2.label);
});

test("Dropdown with avatars closed (mount: enzyme+jsdom)", () => {
    const ddMount = mount(createDropupWithAvatarsAndHorizontalRules(false));
    const dropdownFieldLabel = ddMount
        .find(".dropdown-field-label-group-text")
        .first()
        .text();
    const dropdownFieldAvatar = ddMount
        .find(".avatar")
        .first()
        .prop("src");
    expect(dropdownFieldLabel).toEqual(AVATAR_ELEMENT1.label);
    expect(dropdownFieldAvatar).toEqual(AVATAR_ELEMENT1.avatar);
});

test("Dropdown with avatars open (mount: enzyme+jsdom)", () => {
    const ddMount = mount(createDropupWithAvatarsAndHorizontalRules(true));
    const hrType = ddMount
        .find(".dropdown-elements")
        .children()
        .at(1)
        .type();
    expect(hrType).toEqual("hr");
});
