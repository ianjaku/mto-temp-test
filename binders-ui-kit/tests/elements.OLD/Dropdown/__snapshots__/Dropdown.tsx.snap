// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default dropdown closed (snapshot) 1`] = `
<div
  className="dropdown borders"
  style={
    Object {
      "width": "undefinedpx",
    }
  }
>
  <ul
    className="dropdown-field"
    onBlur={[Function]}
    onClick={[Function]}
    tabIndex={0}
  >
    <li
      className="dropdown-field-label"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="dropdown-field-label-group"
      >
        <span
          className="dropdown-field-label-group-text"
        >
          testtype
        </span>
      </div>
      <span
        className="material-icons dropdown-arrow"
        color=""
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontSize": 24,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      >
        arrow_drop_down
      </span>
    </li>
  </ul>
</div>
`;

exports[`Default dropdown open (snapshot) 1`] = `
<div
  className="dropdown borders dropdown--is-open"
  style={
    Object {
      "width": "undefinedpx",
    }
  }
>
  <ul
    className="dropdown-field"
    onBlur={[Function]}
    onClick={[Function]}
    tabIndex={0}
  >
    <li
      className="dropdown-field-label"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="dropdown-field-label-group"
      >
        <span
          className="dropdown-field-label-group-text"
        >
          testtype
        </span>
      </div>
      <span
        className="material-icons dropdown-arrow"
        color=""
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontSize": 24,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      >
        arrow_drop_down
      </span>
    </li>
  </ul>
  <ul
    className="dropdown-elements"
    style={
      Object {
        "maxHeight": 30000,
      }
    }
  >
    <li
      className="dropdown-field-label"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="dropdown-field-label-group"
      >
        <span
          className="dropdown-field-label-group-text"
        >
          element 1
        </span>
      </div>
    </li>
    <li
      className="dropdown-field-label"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="dropdown-field-label-group"
      >
        <span
          className="dropdown-field-label-group-text"
        >
          element 2
        </span>
      </div>
    </li>
  </ul>
</div>
`;

exports[`dropdown with avatars and horizontal rules (snapshot) 1`] = `
<div
  className="dropdown borders dropdown--is-open"
  style={
    Object {
      "width": "undefinedpx",
    }
  }
>
  <ul
    className="dropdown-field"
    onBlur={[Function]}
    onClick={[Function]}
    tabIndex={0}
  >
    <li
      className="dropdown-field-label contains-avatar"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="dropdown-field-label-group"
      >
        <img
          className="avatar"
          size={30}
          src="http://path.to/image1.jpg"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": "50%",
              "color": "#ffffff",
              "display": "inline-flex",
              "fontSize": 15,
              "height": 30,
              "justifyContent": "center",
              "muiPrepared": true,
              "userSelect": "none",
              "width": 30,
            }
          }
        />
        <span
          className="dropdown-field-label-group-text"
        >
          account 1
        </span>
      </div>
      <span
        className="material-icons dropdown-arrow"
        color=""
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontSize": 24,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      >
        arrow_drop_down
      </span>
    </li>
  </ul>
  <ul
    className="dropdown-elements"
    style={
      Object {
        "maxHeight": 30000,
      }
    }
  >
    <li
      className="dropdown-field-label contains-avatar"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="dropdown-field-label-group"
      >
        <img
          className="avatar"
          size={30}
          src="http://path.to/image1.jpg"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": "50%",
              "color": "#ffffff",
              "display": "inline-flex",
              "fontSize": 15,
              "height": 30,
              "justifyContent": "center",
              "muiPrepared": true,
              "userSelect": "none",
              "width": 30,
            }
          }
        />
        <span
          className="dropdown-field-label-group-text"
        >
          account 1
        </span>
      </div>
    </li>
    <hr />
    <li
      className="dropdown-field-label contains-avatar"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="dropdown-field-label-group"
      >
        <img
          className="avatar"
          size={30}
          src="http://path.to/image2.jpg"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": "50%",
              "color": "#ffffff",
              "display": "inline-flex",
              "fontSize": 15,
              "height": 30,
              "justifyContent": "center",
              "muiPrepared": true,
              "userSelect": "none",
              "width": 30,
            }
          }
        />
        <span
          className="dropdown-field-label-group-text"
        >
          account 2
        </span>
      </div>
    </li>
  </ul>
</div>
`;

exports[`dropdown with selected elemnent (snapshot) 1`] = `
<div
  className="dropdown borders"
  style={
    Object {
      "width": "undefinedpx",
    }
  }
>
  <ul
    className="dropdown-field"
    onBlur={[Function]}
    onClick={[Function]}
    tabIndex={0}
  >
    <li
      className="dropdown-field-label"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="dropdown-field-label-group"
      >
        <span
          className="dropdown-field-label-group-text"
        >
          element 2
        </span>
      </div>
      <span
        className="material-icons dropdown-arrow"
        color=""
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontSize": 24,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      >
        arrow_drop_down
      </span>
    </li>
  </ul>
</div>
`;

exports[`dropup (snapshot) 1`] = `
<div
  className="dropdown borders"
  style={
    Object {
      "width": "undefinedpx",
    }
  }
>
  <ul
    className="dropdown-field"
    onBlur={[Function]}
    onClick={[Function]}
    tabIndex={0}
  >
    <li
      className="dropdown-field-label"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="dropdown-field-label-group"
      >
        <span
          className="dropdown-field-label-group-text"
        >
          testtype
        </span>
      </div>
      <span
        className="material-icons dropdown-arrow"
        color=""
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontSize": 24,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      >
        arrow_drop_down
      </span>
    </li>
  </ul>
</div>
`;
